.github/workflows/deploy-frontend.yaml

```
@@ -0,0 +1,39 @@
+name: deploy-front
+
+on:
+  push:
+    branches:
+      - production
```

```suggestion
      - main
```

I think this one should be main, since the deployment would run when we push a commit to main (which happens after merging a pr)

---

This is intentional to only deploy when new commits are in `production` branch. We shouldn't immediately deploy something just got approve, and deploy should only happen when a release is out.
It has to be strictly isolated between development and production codebase. `main` branch would still be considered as development codebase. In industry, create a release candidate and fully QA on the candidate, then we could deploy the release, with all new features/changes/patches at once.
In smaller scale project like us could skip the release candidate procedure. But consider a feature could be composed by more than one pr, and we shouldn't deploy when a feature is partially done (partial of several PRs have created/merged). Deploy straight from `main` is generally a bad idea in my opinion.

---

ah ok, I didn't know that was intentional. LGTM